source/common/grpc/async_client_impl.cc

```diff
@@ -129,21 +131,18 @@ void AsyncStreamImpl::onData(Buffer::Instance& data, bool end_stream) {
   }
 
   for (auto& frame : decoded_frames_) {
-    ProtobufTypes::MessagePtr response = callbacks_.createEmptyResponse();
-    // TODO(htuch): Need to add support for compressed responses as well here.
-    if (frame.length_ > 0) {
-      Buffer::ZeroCopyInputStreamImpl stream(std::move(frame.data_));
-
-      if (frame.flags_ != GRPC_FH_DEFAULT || !response->ParseFromZeroCopyStream(&stream)) {
-        streamError(Status::GrpcStatus::Internal);
-        return;
-      }
+    if (frame.length_ > 0 && frame.flags_ != GRPC_FH_DEFAULT) {
+      streamError(Status::GrpcStatus::Internal);
+      return;
+    }
+    if (!callbacks_.onReceiveMessageRaw(frame.data_ ? std::move(frame.data_)
+                                                    : std::make_unique<Buffer::OwnedImpl>())) {
+      streamError(Status::GrpcStatus::Internal);
+      return;
     }
-    callbacks_.onReceiveMessageUntyped(std::move(response));
   }
 
-  if (end_stream) {
-    Http::HeaderMapPtr empty_trailers = std::make_unique<Http::HeaderMapImpl>();
+  if (!http_reset_ && end_stream) {
```

Comment:
Can you explain this subtle change in logic?

Reply:
no, reverted